core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2305,6 +2305,23 @@ private[spark] object Utils extends Logging {
     log.info(s"Started daemon with process name: ${Utils.getProcessName()}")
     SignalUtils.registerLogger(log)
   }
+
+  /**
+  * Unions two comma-separated lists of files.
+  */
+  def setUnionFileLists(leftList: String, rightList: String): Set[String] = {
+    val leftSet = if (leftList == "" || leftList == null) {
```

Comment:
If you use `Option[String]` as the argument you can avoid this and the `getOrElse` calls in SparkCOntext, and do something like:

```
var allFiles = Set[String]()

list.foreach { value => allFiles ++= value.split(",") }
```

Which is a lot shorter than the code you have.